components/brave_federated/data_store_service.cc

```diff
@@ -11,48 +11,49 @@
 #include "base/task/thread_pool.h"
 #include "base/threading/sequence_bound.h"
 #include "base/threading/sequenced_task_runner_handle.h"
-#include "brave/components/brave_federated/data_stores/ad_notification_timing_data_store.h"
-
-namespace {
-constexpr char kAdNotificationTaskName[] =
-    "ad_notification_timing_federated_task";
-constexpr int kAdNotificationTaskId = 0;
-constexpr int kMaxNumberOfRecords = 50;
-constexpr int kMaxRetentionDays = 30;
-}  // namespace
+#include "brave/components/brave_federated/data_stores/async_data_store.h"
+#include "brave/components/brave_federated/notification_ad_task_constants.h"
 
 namespace brave_federated {
 
-DataStoreService::DataStoreService(const base::FilePath& database_path)
-    : db_path_(database_path),
-      ad_notification_timing_data_store_(db_path_),
-      weak_factory_(this) {}
+DataStoreService::DataStoreService(const base::FilePath& db_path)
+    : db_path_(db_path), weak_factory_(this) {}
```

Comment:
We could use `base::WeakPtrFactory<DataStoreService> weak_ptr_factory_{this};` in the interface and remove the need for `weak_factory_(this)`